src/main/java/org/mule/extension/email/internal/util/message/SimpleBody.java

```diff
@@ -56,12 +61,25 @@ public SimpleBody(Part part) {
         bodyPart = mp.getBodyPart(0);
         initInlineAttachments(mp);
       } else if (isTextBody(part)) {
-        bodyPart = part;
+        if (part.getDisposition() != null && part.getDisposition().contains(ATTACHMENT)) {
+          inlineAttachments.add(new MessageAttachment(part));
+          bodyPart = new MimeBodyPart();
+          bodyPart.setText("");
```

Comment:
Is this really needed?

Reply:
The logic of the constructors AlternativeBody(bodyPart) and TextBody(bodyPart) expects that bodyPart is not null.
Anyway, I made a refactor and concentrated the empty initialization in a single place.